src/lib/default.nix

```diff
@@ -295,6 +296,17 @@
     then "/${sanitizedRelPath}"
     else sanitizedRelPath;
 
+  systemsFromFile = file:
+    if ! l.pathExists file
+    then
+      throw ''
+        The system for your flake.nix is not initialized yet.
+        Please execute the following command to initialize it:
+
+        nix eval --impure --raw --expr 'builtins.currentSystem' > ./${l.baseNameOf file} && git add ./${l.baseNameOf file}
```

Comment:
This won't work if the user passes a path relative to the `flake.nix` like `./nix/supported-systems`. Maybe we should instead use `config.projectRoot` to strip the prefix path?